app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -34,30 +44,76 @@ class MainActivity : AppCompatActivity() {
         }
 
         val apiRick = retrofit.create(ApiRick::class.java)
-        val titleTextView = findViewById<TextView>(R.id.title)
+        val name = findViewById<TextView>(R.id.character_name)
+        val status = findViewById<TextView>(R.id.character_status)
         val button = findViewById<Button>(R.id.main_button)
         val context = this
+        val displayMetrics = resources.displayMetrics
+        val screenWidth = displayMetrics.widthPixels
+        val imageView = findViewById<ImageView>(R.id.character_image)
+        val desiredSize = (screenWidth * 0.2).toInt()
+        val layout=imageView.layoutParams
+        layout.width = desiredSize
+        layout.height = desiredSize
+        imageView.layoutParams = layout
 
         button.setOnClickListener {
             CoroutineScope(Dispatchers.IO).launch {
                 val response = apiRick.getCharacter()
 
                 if (response.isSuccessful) {
-                    val responseBody = response.body()
-                    runOnUiThread {
-                        titleTextView.text = responseBody.toString()
+                    val characterResponse = response.body()
+                    val firstCharacter = characterResponse?.results?.firstOrNull()
+                    firstCharacter?.let { character ->
+                        withContext(Dispatchers.Main) {
+                            name.text = character.name?.let {
```

Comment:
is this ? needed? character is not null and name is a mandatory param of the character.